WebDriverAgentLib/Categories/XCUIElement+FBScrolling.m

```diff
@@ -279,4 +281,23 @@ - (BOOL)fb_scrollAncestorScrollViewByVectorWithinScrollViewFrame:(CGVector)vecto
   return didSucceed;
 }
 
+- (XCElementSnapshot *)fb_findVisibleScrollableView
+{
+  NSArray *possibleParents = @[
+                               @(XCUIElementTypeScrollView),
+                               @(XCUIElementTypeCollectionView),
+                               @(XCUIElementTypeTable),
+                              ];
+    
+  XCElementSnapshot *scrollView = [self fb_parentMatchingOneOfTypes:possibleParents];
+    
+  if (scrollView == nil) {
+    return nil;
+  } else if (![scrollView fb_isVisible]) {
+    return [scrollView fb_findVisibleScrollableView];
+  } else {
```

Comment:
I can move this logic to `fb_parentMatchingOneOfTypes` as well by renaming it to `fb_findVisiableParentMatchingOneOfTypes`